pkg/controller/datavolume-controller.go

```diff
@@ -1222,22 +1377,76 @@ func (r *DatavolumeReconciler) snapshotSmartClonePossible(dataVolume *cdiv1.Data
 	return true, nil
 }
 
-func (r *DatavolumeReconciler) getCloneStrategy() (cdiv1.CDICloneStrategy, error) {
+func (r *DatavolumeReconciler) getCloneStrategy(dataVolume *cdiv1.DataVolume) (*cdiv1.CDICloneStrategy, error) {
+	defaultCloneStrategy := cdiv1.CDICloneStrategy(cdiv1.CloneStrategySnapshot)
+	sourcePvc, err := r.findSourcePvc(dataVolume)
+	if err != nil {
+		// when k8serrors.IsNotFound(err) -> Source PVC is missing, cannot correctly select strategy
```

Comment:
this is a place where new logic differs from the old one, 
- previously when the `source` pvc was not available clone strategy was assumed as HostAssisted and the reconcile loop logic moved forward and reconciled the `target` pvc 
- with new logic it returns err, I wonder if some kind of event might be helpful here?

Reply:
So the err will result in a new reconcile and we hope that in that time the source appeared, also if there is no source, host assisted clone will also not work, since there is no source. I think an event is probably a good idea, something like source not found unable to clone or something like that.

Reply:
Yes, I think it is only good  when `k8serrors.IsNotFound(err)` other errors are just an intermittent communication problems or the whole cluster is bad.